doc/dynamic-environments/configuration.mkd

```diff
@@ -356,6 +356,29 @@ sources:
 * if `false` (default) environment folder will not be prefixed
 * if `String` environment folder will be prefixed with the `prefix` value.
 
+### branch_filter
+
+The 'branch_filter' setting restricts the environment that are deployed to only
+those that match the given filter. The filter is a regex string. Each branch in
+the 'git' repo will have its named tested against this regex and, if the regex
+matches the branch name, then an environment will be deployed for this branch.
+If no 'branch_filter' is specified, then all branches in the 'git' repo will
+be deployed (default behavior).
+
+#### branch_filter behaviour
+* if empty, deploy environments for all branches
+* for each branch in git repo
+** if `branch.name` matches `branch_filter` regex, then deploy an environment for branch
+
+Example: only deploy branches named 'prod' or 'qa'
+```yaml
+---
+sources:
+  mysource:
+    basedir: '/etc/puppet/environments'
+    branch_filter: '^(prod)|(qa)$'
```

Comment:
You shouldn't need 2 groups here. `'^(prod|qa)'` should also work?

Reply:
Yes, but it also doesn't matter. We're not actually using the group information so having 1 or 2 groups makes no difference.